app/gilded-rose.ts

```diff
@@ -17,53 +19,65 @@ export class GildedRose {
+      if (this.items[i].name.toLowerCase().indexOf('sulfuras') === -1) {
+        this.items[i].sellIn -= 1
+      }
+
+      let qualityChange = 0;
+      if (this.items[i].name == 'Aged Brie' || this.items[i].name.toLowerCase().includes('backstage passes'))  {
+        // Quality Increases
+        qualityChange = 1
+
+        if (this.items[i].name.toLowerCase().includes('backstage passes')) {
+          // Backstage Pass Quality depends on age
+          if (this.items[i].sellIn <= 10) {
+            qualityChange += 1;
           }
-        }
-      } else {
-        if (this.items[i].quality < 50) {
-          this.items[i].quality = this.items[i].quality + 1
-          if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].sellIn < 11) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
-            if (this.items[i].sellIn < 6) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
+
+          if (this.items[i].sellIn <= 5) {
+            qualityChange += 1;
+          }
+
+          if (this.items[i].sellIn < 0) {
+            qualityChange = -1 * this.items[i].quality;
           }
         }
+
+        // Caps at 50
+        if (this.items[i].quality + qualityChange > 50) {
+          qualityChange = 50 - this.items[i].quality;
+        }
+
+      } else if (this.items[i].name.toLowerCase().indexOf('sulfuras') === -1) {
+        // Quality Decreases
+        qualityChange = -1;
+
+      } else {
+        // Sulfuras quality remains constant
+        qualityChange = 0;
       }
-      if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-        this.items[i].sellIn = this.items[i].sellIn - 1;
-      }
+
       if (this.items[i].sellIn < 0) {
-        if (this.items[i].name != 'Aged Brie') {
-          if (this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].quality > 0) {
-              if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-                this.items[i].quality = this.items[i].quality - 1
-              }
-            }
-          } else {
-            this.items[i].quality = this.items[i].quality - this.items[i].quality
-          }
-        } else {
-          if (this.items[i].quality < 50) {
-            this.items[i].quality = this.items[i].quality + 1
-          }
-        }
+        // Negative SellIn doubles QC
+        qualityChange *= 2;
+      }
+
+      if (this.items[i].name.toLowerCase().includes('conjured')) {
+        // Conjured doubles QC
+        qualityChange *= 2;
       }
-    }
 
+      this.items[i].quality += qualityChange;
+      if (this.items[i].quality < 0) {
+        // Caps Quality at 0
+        this.items[i].quality = 0;
+      }
+    }
```

Comment:
This is a lot cleaner than the original - but I think it can be even cleaner if you separated out some parts of the code into other functions. E.g. one for aged brie, one for conjured, etc.